File: t/PACSFramework/VB/DicomHighLevelMWLScuDemo/MppsIodInfo.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Leadtools.Dicom.Common.Extensions
Imports Leadtools.Dicom
Imports Leadtools.Dicom.Common.DataTypes

Namespace DicomDemo
   Public Class MppsIodInfo
      Private _StudyInstanceUID As String

      <Element(DicomTag.StudyInstanceUID)> _
      Public Property StudyInstanceUID() As String
         Get
            Return _StudyInstanceUID
         End Get
         Set(ByVal value As String)
            _StudyInstanceUID = Value
         End Set
      End Property

      Private _ReferencedStudySequence As List(Of SopInstanceReference)

      <Element(DicomTag.ReferencedStudySequence)> _
      Public Property ReferencedStudySequence() As List(Of SopInstanceReference)
         Get
            Return _ReferencedStudySequence
         End Get
         Set(ByVal value As List(Of SopInstanceReference))
            _ReferencedStudySequence = Value
         End Set
      End Property

      Private _AccessionNumber As String

      <Element(DicomTag.AccessionNumber)> _
      Public Property AccessionNumber() As String
         Get
            Return _AccessionNumber
         End Get
         Set(ByVal value As String)
            _AccessionNumber = Value
         End Set
      End Property

      Private _RequestAttributesSequence As List(Of RequestAttributes)

      <Element(DicomTag.RequestAttributesSequence)> _
      Public Property RequestAttributesSequence() As List(Of RequestAttributes)
         Get
            Return _RequestAttributesSequence
         End Get
         Set(ByVal value As List(Of RequestAttributes))
            _RequestAttributesSequence = Value
         End Set
      End Property

      Private _StudyID As String

      <Element(DicomTag.StudyID)> _
      Public Property StudyID() As String
         Get
            Return _StudyID
         End Get
         Set(ByVal value As String)
            _StudyID = Value
         End Set
      End Property

      Private _PerformedProcedureStepID As String

      <Element(DicomTag.PerformedProcedureStepID)> _
      Public Property PerformedProcedureStepID() As String
         Get
            Return _PerformedProcedureStepID
         End Get
         Set(ByVal value As String)
            _PerformedProcedureStepID = Value
         End Set
      End Property

      Private _PerformedProcedureStepStartDate As Nullable(Of DateTime)

      <Element(DicomTag.PerformedProcedureStepStartDate)> _
      Public Property PerformedProcedureStepStartDate() As Nullable(Of DateTime)
         Get
            Return _PerformedProcedureStepStartDate
         End Get
         Set(ByVal value As Nullable(Of DateTime))
            _PerformedProcedureStepStartDate = Value
         End Set
      End Property

      Private _PerformedProcedureStepStartTime As Nullable(Of DateTime)

      <Element(DicomTag.PerformedProcedureStepStartTime)> _
      Public Property PerformedProcedureStepStartTime() As Nullable(Of DateTime)
         Get
            Return _PerformedProcedureStepStartTime
         End Get
         Set(ByVal value As Nullable(Of DateTime))
            _PerformedProcedureStepStartTime = Value
         End Set
      End Property

      Private _PerformedProcedureStepDescription As String

      <Element(DicomTag.PerformedProcedureStepDescription)> _
      Public Property PerformedProcedureStepDescription() As String
         Get
            Return _PerformedProcedureStepDescription
         End Get
         Set(ByVal value As String)
            _PerformedProcedureStepDescription = Value
         End Set
      End Property

      Private _ProcedureCodeSequence As List(Of CodeSequence) = New List(Of CodeSequence)()

      <Element(DicomTag.ProcedureCodeSequence)> _
      Public Property ProcedureCodeSequence() As List(Of CodeSequence)
         Get
            Return _ProcedureCodeSequence
         End Get
         Set(ByVal value As List(Of CodeSequence))
            _ProcedureCodeSequence = Value
         End Set
      End Property

      Private _ReferencedPerformedProcedureStepSequence As List(Of SopInstanceReference)

      <Element(DicomTag.ReferencedPerformedProcedureStepSequence)> _
      Public Property ReferencedPerformedProcedureStepSequence() As List(Of SopInstanceReference)
         Get
            Return _ReferencedPerformedProcedureStepSequence
         End Get
         Set(ByVal value As List(Of SopInstanceReference))
            _ReferencedPerformedProcedureStepSequence = Value
         End Set
      End Property

      Private _ProtocolName As String

      <Element(DicomTag.ProtocolName)> _
      Public Property ProtocolName() As String
         Get
            Return _ProtocolName
         End Get
         Set(ByVal value As String)
            _ProtocolName = Value
         End Set
      End Property
   End Class
End Namespace

File: t/PACSFramework/CS/DicomHighLevelMWLScuDemo/MppsIodInfo.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Text;
using Leadtools.Dicom.Common.Extensions;
using Leadtools.Dicom;
using Leadtools.Dicom.Common.DataTypes;

namespace DicomDemo
{
    public class MppsIodInfo
    {
        private string _StudyInstanceUID;

        [Element(DicomTag.StudyInstanceUID)]
        public string StudyInstanceUID
        {
            get { return _StudyInstanceUID; }
            set { _StudyInstanceUID = value; }
        }

        private List<SopInstanceReference> _ReferencedStudySequence;
        
        [Element(DicomTag.ReferencedStudySequence)]
        public List<SopInstanceReference> ReferencedStudySequence
        {
            get { return _ReferencedStudySequence; }
            set { _ReferencedStudySequence = value; }
        }

        private string _AccessionNumber;

        [Element(DicomTag.AccessionNumber)]
        public string AccessionNumber
        {
            get { return _AccessionNumber; }
            set { _AccessionNumber = value; }
        }

        private List<RequestAttributes> _RequestAttributesSequence;

        [Element(DicomTag.RequestAttributesSequence)]
        public List<RequestAttributes> RequestAttributesSequence
        {
            get { return _RequestAttributesSequence; }
            set { _RequestAttributesSequence = value; }
        }

        private string _StudyID;

        [Element(DicomTag.StudyID)]
        public string StudyID
        {
            get { return _StudyID; }
            set { _StudyID = value; }
        }

        private string _PerformedProcedureStepID;

        [Element(DicomTag.PerformedProcedureStepID)]
        public string PerformedProcedureStepID
        {
            get { return _PerformedProcedureStepID; }
            set { _PerformedProcedureStepID = value; }
        }

        private DateTime? _PerformedProcedureStepStartDate;

        [Element(DicomTag.PerformedProcedureStepStartDate)]
        public DateTime? PerformedProcedureStepStartDate
        {
            get { return _PerformedProcedureStepStartDate; }
            set { _PerformedProcedureStepStartDate = value; }
        }

        private DateTime? _PerformedProcedureStepStartTime;

        [Element(DicomTag.PerformedProcedureStepStartTime)]
        public DateTime? PerformedProcedureStepStartTime
        {
            get { return _PerformedProcedureStepStartTime; }
            set { _PerformedProcedureStepStartTime = value; }
        }

        private string _PerformedProcedureStepDescription;

        [Element(DicomTag.PerformedProcedureStepDescription)]
        public string PerformedProcedureStepDescription
        {
            get { return _PerformedProcedureStepDescription; }
            set { _PerformedProcedureStepDescription = value; }
        }

        private List<CodeSequence> _ProcedureCodeSequence = new List<CodeSequence>();
        
        [Element(DicomTag.ProcedureCodeSequence)]
        public List<CodeSequence> ProcedureCodeSequence
        {
            get { return _ProcedureCodeSequence; }
            set { _ProcedureCodeSequence = value; }
        }

        private List<SopInstanceReference> _ReferencedPerformedProcedureStepSequence;

        [Element(DicomTag.ReferencedPerformedProcedureStepSequence)]
        public List<SopInstanceReference> ReferencedPerformedProcedureStepSequence
        {
            get { return _ReferencedPerformedProcedureStepSequence; }
            set { _ReferencedPerformedProcedureStepSequence = value; }
        }

        private string _ProtocolName;

        [Element(DicomTag.ProtocolName)]
        public string ProtocolName
        {
            get { return _ProtocolName; }
            set { _ProtocolName = value; }
        }
    }
}
